### PR TITLE
 backupccl: add RestoreData processor

### PR DESCRIPTION
**Pull request by pbardea**
This commit adds a processor which actually performs the ImportRequest.
It has an input which accepts rows with 2 columns that should be sent
from SplitAndScatter processors. Each row represents one span that the
processor should ingest. The intention is that the spans direcected to a
processor on a given node have their leaseholder colocated on the same
node (this work is done in the SplitAndScatter processor). All that
remains is to send a request to ingest the data and stream back its
progress to the coordinator upon completion.

Part of #40239.

Release note: None

### CODE REVIEW

**Comment by pbardea** [comment]
Friendly ping.

File: pkg/ccl/backupccl/restore_data_processor.go [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+		}
+		if row == nil {
+			// Done.
+			break
+		}
+
+		if len(row) != 2 {
+			return errors.New("expected input rows to have exactly 2 columns")
+		}
+		alloc := &sqlbase.DatumAlloc{}
```
Comment (dt) [inline pkg/ccl/backupccl/restore_data_processor.go:120]: nit: could pull this out of the loop so it gets reused as intended.

**Review by dt — APPROVED** [review]

File: pkg/sql/execinfrapb/processors_bulk_io.proto [hunk 2]
```diff
@@ -161,6 +161,17 @@ message BackupDataSpec {
 message RestoreSpanEntry {
   optional roachpb.Span span = 1 [(gogoproto.nullable) = false];
   repeated roachpb.ImportRequest.File files = 2 [(gogoproto.nullable) = false];
+  optional int64 progressIdx = 3 [(gogoproto.nullable) = false];
```
Comment (dt) [inline pkg/sql/execinfrapb/processors_bulk_io.proto:164]: what sets this?

Comment (pbardea) [reply pkg/sql/execinfrapb/processors_bulk_io.proto:164]: The coordinator will set this in `restore_job` - it's currently not set right now, but will be in the planning PR.

File: pkg/ccl/backupccl/restore_data_processor.go (hunk 1 above)
Comment (pbardea) [reply pkg/ccl/backupccl/restore_data_processor.go]: Done.

**Comment by pbardea** [comment]
TFTR!
bors r=dt

**Pull request merged**